R/dummy_data.R

```diff
@@ -1330,3 +1330,278 @@ dummy_er <- function(nid = 1000, n_hospitals = 10, time_period = c(2015, 2023),
   # return only with required columns
   return(df1[order(df1$genc_id), c("genc_id", "hospital_num", "triage_date_time")])
 }
+
+#' @title
+#' Generate simulated radiology data
+#'
+#' @description
+#' This function creates a dummy dataset with a subset of variables that
+#' are contained in the GEMINI "radiology" table, as seen in
+#' [GEMINI Data Repository Dictionary](https://geminimedicine.ca/the-gemini-database/).
+#'
+#' This function only simulates modalities used in Our Practice Report (OPR) - CT, MRI,
+#' Ultrasound. It does not cover all modalities seen in the actual "radiology" data table.
+#'
+#' @param nid (`integer`)\cr Number of unique encounter IDs to simulate.
+#' Encounter IDs may repeat to simulate multiple radiology tests,
+#' resulting in a data table with more rows than `nid`.
+#' This input is optional if `cohort` is provided.
+#'
+#' @param n_hospitals(`integer`)\cr The number of hospitals to simulate.
+#' This optional is optional if `cohort` is provided.
+#'
+#' @param time_period (`vector`)\cr A numeric or character vector containing the data range of the data
+#' by years or specific dates in either format: ("yyyy-mm-dd", "yyyy-mm-dd") or (yyyy, yyyy)
+#' The start date and end date will be (yyyy-01-01 and yyyy-12-31) if (yyyy, yyyy)
+#' is the date range format provided. Optional when `cohort` is provided.
+#'
+#' @param cohort  (`data.frame or data.table`)\cr Optional, data frame or data table with the following columns:
+#' - `genc_id` (`integer`): Mock encounter ID number
+#' - `hospital_num` (`integer`): Mock hospital ID number
+#' - `admission_date_time` (`character`): Date and time of IP admission in YYYY-MM-DD HH:MM format
+#' - `discharge_date_time` (`character`): Date and time of IP discharge in YYYY-MM-DD HH:MM format.
+#' When `cohort` is not NULL, `nid`, `n_hospitals`, and `time_period` are ignored.
+#'
+#' @param seed (`integer`)\cr Optional, a number to be used to set the seed for reproducible results.
+#'
+#' @return (`data.table`)\cr A `data.table` object similar that contains the following fields:
+#' - `genc_id` (`integer`): Mock encounter ID number; integers starting from 1 or from `cohort` if provided
+#' - `hospital_id` (`integer`): Mock hospital ID number; integers starting from 1 or from `cohort` if provided
```

Comment:
I believe it always returns `hospital_num`?
```suggestion
#' - `hospital_num` (`integer`): Mock hospital ID number; integers starting from 1 or from `cohort` if provided
```